script_facade/api/fhir.py

```diff
@@ -8,39 +8,58 @@
 blueprint = Blueprint('fhir', __name__)
 
 
+def required_search_request_params(req, context):
+    """Extract common request parameters for search APIs
+
+    :param req: the live request
+    :param context: calling context used in exception text
+    :returns: dictionary of required and optional search parameters
+    :raises: BadRequest if any required parameters are missing
+
+    """
+    patient_fname = req.args.get('subject:Patient.name.given')
+    patient_lname = req.args.get('subject:Patient.name.family')
+    patient_dob = req.args.get('subject:Patient.birthdate', '').split('eq')[-1]
+    dea = req.args.get('dea')
+
+    if not all((patient_fname, patient_lname, patient_dob, dea)):
+        current_app.logger.warning(
+            "%s search attempted without all required parameters"
+            "{fname: %s, lname: %s, dob: %s, dea: %s",
+            context, patient_fname, patient_lname, patient_dob, dea)
+        return 'Required parameters not given', 400
```

Comment:
Should this return JSON?
```suggestion
        return jsonify(message='Required parameters not given'), 400
```